app/src/test/java/org/simple/clinic/settings/SettingsEffectHandlerTest.kt

```diff
@@ -0,0 +1,76 @@
+package org.simple.clinic.settings
+
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Observable
+import io.reactivex.observers.TestObserver
+import io.reactivex.subjects.PublishSubject
+import org.junit.After
+import org.junit.Test
+import org.simple.clinic.patient.PatientMocker
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.None
+import org.simple.clinic.util.scheduler.TrampolineSchedulersProvider
+import org.simple.clinic.util.toOptional
+import java.util.UUID
+
+class SettingsEffectHandlerTest {
+
+  private val userSession = mock<UserSession>()
+  private val effectsSubject = PublishSubject.create<SettingsEffect>()
+
+  private val effectHandlerTest: TestObserver<SettingsEvent> = effectsSubject
+      .compose(SettingsEffectHandler.create(
+          userSession = userSession,
+          schedulersProvider = TrampolineSchedulersProvider()
+      ))
+      .test()
+
+  @After
+  fun tearDown() {
+    effectHandlerTest.dispose()
+  }
+
+  @Test
+  fun `user details must be fetched when the load user details event is received`() {
+    // given
+    val firstUser = PatientMocker.loggedInUser(
+        uuid = UUID.fromString("eadfd8ef-6c88-4bc9-ba96-e05bec2d6d8b"),
+        name = "Mahalakshmi Puri",
+        phone = "1234567890"
+    )
+    val secondUser = PatientMocker.loggedInUser(
+        uuid = UUID.fromString("522f684d-d1a8-4bbb-9ef5-80e82369f061"),
+        name = "Anish Acharya",
+        phone = "0987654321"
+    )
+    whenever(userSession.loggedInUser()).thenReturn(Observable.just(firstUser.toOptional(), secondUser.toOptional()))
+
+    // when
+    effectsSubject.onNext(LoadUserDetailsEffect)
+
+    // then
+    effectHandlerTest
+        .assertValues(
+            UserDetailsLoaded(name = firstUser.fullName, phoneNumber = firstUser.phoneNumber),
+            UserDetailsLoaded(name = secondUser.fullName, phoneNumber = secondUser.phoneNumber)
+        )
+        .assertNotComplete()
+        .assertNotTerminated()
+  }
+
+  @Test
+  fun `when there are no user details, no event must be triggered`() {
```

Comment:
If there are not user details, then isn't this failing silently? The prerequisite for this screen is that, there should be a logged in user. Otherwise we won't land in this screen. Am I missing something here?

Reply:
We have a logout feature that will clear the local logged in user if they get signed out in the background. This test is to ensure that the app will not crash if this screen is open when that happens,

Reply:
Done.